### ISSUE TITLE
Feature: Highlight currently selected layer in grid mode

### ISSUE THREAD

**Issue by tlambert03** [napari/napari#2954]
## 🚀 Feature
In grid mode, with enough layers, it can be hard to know which image corresponds to which layer...

It would be nice to have a faint outline on the currently selected image(s), as shown in this mockup:

<img width="976" alt="Untitled" src="https://user-images.githubusercontent.com/1609449/123856316-f1793280-d8ee-11eb-825e-1696833e61f0.png">

**Comment by goanpeca** [comment]
Pretty cool, would that be like a toggle or when hovering the layer list or always highlighted?

**Comment by tlambert03** [comment]
TBD!

@sofroniewn?  @jni?

could also be a quick flash animation when you change the selection?

**Comment by goanpeca** [comment]
I think it would be nice if you could select multiple layers on the list control and have them highlight as you keep selecting more :)

**Comment by tlambert03** [comment]
ooo.  yeah!

ok, so at least one mode in which they remain highlighted as long as they are in `layer.selection` would be nice, yeah?

**Comment by sofroniewn** [comment]
I think this could leverage a newly added highlight visual - see #2560 for some discussion of other possible use cases. I'd also maybe use that visual to remove highlighting from the points and shapes layers themselves (which are compound visuals). I know @alisterburt and @kevinyamauchi have need for a highlight visual to help interactivity with some the 3D clipping plane stuff. Plugins could also then get access to this highlight visual (which I see as a little more akin to things like our scale bar etc.) A very simple one would just be a bounding box highlight that we could reuse for any type of selection, points, shapes etc. Could let people drag to zoom into a region, or drag to crop - things that for now people use the rectangle in the shapes layer for sometimes.

So basically - yes, I think this would be great! But there's maybe a bigger discussion about what this could be in its full potential.

@tlambert03 is your mockup in code? Are you working on this/ interesting in working on this or is this something that might go in the sprint queue? (either the small feature, or bigger picture vision for highlighting)

**Comment by jni** [comment]
Love this! And yes I do like some kind of visual feedback in the canvas on hover as suggested by @goanpeca

**Comment by tlambert03** [comment]
> I think this could leverage a newly added highlight visual - see #2560 for some discussion of other possible use cases.

Oh absolutely, I had forgotten about that one.  Definitely, this could leverage a more generic "highlight this part of the canvas" API.

**Comment by tlambert03** [comment]
> @tlambert03 is your mockup in code? Are you working on this/ interesting in working on this or is this something that might go in the sprint queue?

Nope, not in code, just drew a box on a screenshot :) I'd love for this to go in the sprint queue.

**Comment by brisvag** [comment]
I agree that it would be great to have a "HUD" system in a more generalized form for interactive stuff.

Things where this could be useful:
- click between 2 points to measure a distance (don't want to create a points layer nor a vector layer to display the distance path, for example)
- highlight a draggable plane (see [this video of @kevinyamauchi and @alisterburt 's cropping plugin](#https://github.com/napari/napari/pull/2437#issuecomment-822371530)

**Comment by VolkerH** [comment]
I have exactly the same use case, so it would be nice to have such a feature as suggested in the original post.

In addition, I could see a use where layers can be selected by clicking into the grid, e.g. to select images on which a plugin should operate. 
However that would again be a different mouse/selection mode from the usual image navigation so maybe too confusing for users.

**Comment by kevinyamauchi** [comment]
@brisvag , do you think #4849 would be useful here?

**Comment by brisvag** [comment]
Exactly!

**Comment by psobolewskiPhD** [comment]
Something like `show bounding_box for selected layer` would work quite nicely for this!
